### PR TITLE
1447 1445 list all project experiment organization response not intuitive

### PR DESCRIPTION
**Pull request by lusnig**
tickets:
1. https://terraquantum.atlassian.net/browse/TQ42-1445
2. https://terraquantum.atlassian.net/browse/TQ42-1447

I implemented the `__repr__` methods and now the system prints out the list of organisations, experiments and projects in an human readable manner as described in the [documentation](https://terra-quantum-tq42sdk-docs.readthedocs-hosted.com/en/latest/Python_Developer_Guide/Setting_Up_Your_Environment.html#list-all-organizations)

### CODE REVIEW

File: notebooks/ts_generic_example.ipynb [hunk 1]
```diff
@@ -296,6 +297,7 @@
     "            Layer(classical_dense_layer=ClassicalDenseLayer(hidden_size=1, bias=False)),\n",
     "        ],\n",
     "        num_epochs=5,\n",
+    "        k_fold=1,\n",
```
Comment (sbtq) [inline notebooks/ts_generic_example.ipynb:300]: if you are also adding k_fold here. Please then also upgrade the grpc client version from the dependencies.

Comment (lusnig) [reply notebooks/ts_generic_example.ipynb:300]: done

File: tq42/compute.py [hunk 2]
```diff
@@ -36,6 +36,9 @@ def __init__(self, hardware: HardwareProto = HardwareProto.HARDWARE_UNSPECIFIED)
             gpu=configuration.get("GPU", ""),
         )
 
+    def __repr__(self):
+        return f"(Hardware name: { self.config.name}, Cpu: { self.config.cpu}, Gpu: { self.config.gpu}, Memory: { self.config.memory}, Storage: { self.config.storage})"
```
Comment (sbtq) [inline tq42/compute.py:40]: Can we please do some proper formatting for the variables 🙈 These spaces are freaking me out a bit 😄

Comment (sbtq) [inline tq42/compute.py:40]: do we have to manually construct this string or couldn't we just join all of the key values from the `HardwareConfig` via `self.config`?

File: tq42/dataset.py [hunk 3]
```diff
@@ -39,6 +39,10 @@ def __init__(
         else:
             self.data = self._get()
 
+    def __repr__(self):
+        created_at = str(self.data.created_at).replace("\n", " ")
+        return f"(Dataset ID: { self.data.id}, Name: { self.data.name}, Project ID: { self.data.project_id}, Sensitivity: { self.data.sensitivity}, Created by: { self.data.created_by}, Created at: {created_at })"
```
Comment (sbtq) [inline tq42/dataset.py:44]: doing it like this feels really brittle. Do you go in and manually adjust all of the `__repr__` functions when we actually add new fields?

Comment (sbtq) [inline tq42/dataset.py:44]: Please at some point properly align the variables with spaces or without spaces. This is really triggering my inner monk 😆

**Review by sbtq — CHANGES_REQUESTED** [review]
Not consistent and no proper formatting, please re-do some of the stuff.

**Review by rbattq — CHANGES_REQUESTED** [review]
According to official [docs](https://docs.python.org/3/library/functions.html#repr) 

"repr(object)
Return a string containing a printable representation of an object. For many types, this function makes an attempt to return a string that would yield an object with the same value when passed to eval()..."

This means that if possible, `__repr__()` should return a string which could be evaluated to produce and object of the corresponding class.

File: tq42/compute.py (hunk 2 above)
Comment (lusnig) [reply tq42/compute.py]: done it using dynamic data trasformation approach

Comment (lusnig) [reply tq42/compute.py]: done it using dynamic data trasformation approach

File: tq42/dataset.py (hunk 3 above)
Comment (lusnig) [reply tq42/dataset.py]: done it using dynamic data trasformation approach

[the remark this answers is not in the document]
Comment (lusnig) [reply tq42/dataset.py]: done it using dynamic data trasformation approach and we print all the internal value as in the `__str__` method

Comment (lusnig) [reply tq42/dataset.py]: done it using dynamic data trasformation approach

File: tq42/experiment.py [hunk 4]
```diff
@@ -48,6 +48,9 @@ def __init__(
         else:
             self.data = self._get_data()
 
+    def __repr__(self):
+        return f"(Experiment ID: { self.data.id}, Name: { self.data.name})"
```
[the remark this answers is not in the document]
Comment (lusnig) [reply tq42/experiment.py]: done it using dynamic data trasformation approach

**Comment by lusnig** [comment]
@rbattq I spoke with @sbtq.
We would like for now improve the user experience rather than chasing technical ideals.
Are you also ok with that?

**Review by sbtq — CHANGES_REQUESTED** [review]
This implementation is not at all what we talked about.....
Either you go ahead and take everything in the `self.__dict__` or you do a shorter version just for the `__repr__` which I told you is my preferred way currently with just giving out id and name together with the class type. You did not go either of those directions....

**Comment by lusnig** [comment]
__repr__ returns less infos

File: tq42/experiment_run.py [hunk 5]
```diff
@@ -51,6 +53,11 @@ def __init__(
     def __str__(self):
         return str(self.data)
 
+    def __repr__(self):
+        print("LUCA")
```
Comment (rbattq) [inline tq42/experiment_run.py:57]: What?

Comment (lusnig) [reply tq42/experiment_run.py]: removed

**Pull request closed without merging**